src/server/terrain/biomes.zig

```diff
@@ -20,7 +20,7 @@ pub const SimpleStructureModel = struct { // MARK: SimpleStructureModel
 		water_surface,
 	};
 	const VTable = struct {
-		loadModel: *const fn(parameters: ZonElement) *anyopaque,
+		loadModel: *const fn(parameters: ZonElement) ?*anyopaque,
```

Comment:
Although compiler seems to ignore it, I think it is unsafe to use `?*anyopaque` here without changing the return type in all structure implementation. Current implementation can support it, but its definitely not the part of the contract. Even if it would be completely safe, for sake of consistency return type in every implementation of `loadModel` should be optional.

Reply:
Updated

Reply:
Could you then please also adjust the corresponding casting functions (utils.castFunctionReturnToAnyopaque and utils.castFunctionSelfToAnyopaque) so they do throw a compiler error on optional mismatch.

Reply:
I think I'm piecing it together.

You want a compiler error thrown if something is added that implements `loadModel` but it tries to return a non-optional `anyopaque`

Reply:
The simplest solution seems to be to create a new function for optionals.

I can't seem to figure out a cleaner way that keeps everything in one generalized function.

Reply:
Okay I pushed up an attempt at this. It feels wrong, though.